kie-server-parent/kie-server-api/pom.xml

```diff
@@ -156,6 +175,13 @@
           </instructions>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <configuration>
+          <failOnViolation>true</failOnViolation>
```

Comment:
I think this can be ommitted by overriding property from kie-parent:
`<checkstyle.failOnViolation>false</checkstyle.failOnViolation>`

This will turn on failOnViolation too and it is much shorter.